src/test/jtx/Oracle.h

```diff
@@ -0,0 +1,174 @@
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_TEST_JTX_ORACLE_H_INCLUDED
+#define RIPPLE_TEST_JTX_ORACLE_H_INCLUDED
+
+#include <test/jtx.h>
+
+namespace ripple {
+namespace test {
+namespace jtx {
+
+// base asset, quote asset, price, scale
+using DataSeries = std::vector<std::tuple<
+    std::string,
+    std::string,
+    std::optional<std::uint32_t>,
+    std::optional<std::uint8_t>>>;
+
+// Typical defaults for Create
+struct CreateArg
+{
+    std::optional<AccountID> owner = std::nullopt;
+    std::optional<std::uint32_t> documentID = 1;
+    DataSeries series = {{"XRP", "USD", 740, 1}};
+    std::optional<std::string> assetClass = "currency";
+    std::optional<std::string> provider = "provider";
+    std::optional<std::string> uri = "URI";
+    std::optional<std::uint32_t> lastUpdateTime = std::nullopt;
+    std::uint32_t flags = 0;
+    std::optional<jtx::msig> msig = std::nullopt;
+    std::optional<jtx::seq> seq = std::nullopt;
+    std::uint32_t fee = 10;
+    std::optional<ter> err = std::nullopt;
+};
+
+// Typical defaults for Update
+struct UpdateArg
+{
+    std::optional<AccountID> owner = std::nullopt;
+    std::optional<std::uint32_t> documentID = std::nullopt;
+    DataSeries series = {};
+    std::optional<std::string> assetClass = std::nullopt;
+    std::optional<std::string> provider = std::nullopt;
+    std::optional<std::string> uri = "URI";
+    std::optional<std::uint32_t> lastUpdateTime = std::nullopt;
+    std::uint32_t flags = 0;
+    std::optional<jtx::msig> msig = std::nullopt;
+    std::optional<jtx::seq> seq = std::nullopt;
+    std::uint32_t fee = 10;
+    std::optional<ter> err = std::nullopt;
+};
+
+struct RemoveArg
+{
+    std::optional<AccountID> const& owner = std::nullopt;
+    std::optional<std::uint32_t> const& documentID = std::nullopt;
+    std::optional<jtx::msig> const& msig = std::nullopt;
+    std::optional<jtx::seq> seq = std::nullopt;
+    std::uint32_t fee = 10;
+    std::optional<ter> const& err = std::nullopt;
+};
+
+/** Oracle class facilitates unit-testing of the Price Oracle feature.
+ * It defines functions to create, update, and delete the Oracle object,
+ * to query for various states, and to call APIs.
+ */
+class Oracle
```

Comment:
I like the use of this class.

Reply:
Thanks. I find it quite useful to have a class that handles the tests rather than a collection of individual functions. I believe it makes the implementation of unit tests a bit easier and more streamlined.